src/main/java/paymelah/model/debt/Money.java

```diff
@@ -29,7 +33,24 @@ public Money(String money) {
         if (money.startsWith("$")) {
             money = money.substring(1);
         }
-        value = new BigDecimal(money).setScale(CENTS_PRECISION);
+
+        int plusCount = 0;
+        while (money.endsWith("+")) {
+            plusCount += 1;
+            money = money.substring(0, money.length() - 1).trim();
+        }
+
+        BigDecimal value = new BigDecimal(money).setScale(CENTS_PRECISION);
+        if (plusCount == 1) {
+            value = value.multiply(GST_MULTIPLIER).setScale(CENTS_PRECISION, RoundingMode.CEILING);
+        } else if (plusCount == 2) {
+            value = value.multiply(SERVICE_CHARGE_MULTIPLIER).multiply(GST_MULTIPLIER)
+                    .setScale(CENTS_PRECISION, RoundingMode.CEILING);
+        } else if (plusCount != 0) {
+            assert false : ">2 '+' not detected as invalid Money input.";
```

Comment:
Should this be an assert? This is not a developer error, maybe we should throw a ParseException instead

Reply:
Never mind, it should not be able to reach here with more than 2 +s

Reply:
The full code (hidden by the diff) already checks if the `money:String` is valid and throws an exception if it is not, so if it enters this if block it should be a dev error.